### ISSUE TITLE
Implement keyboard accessibility

### ISSUE THREAD

**Issue by oliviertassinari** [storybookjs/react-inspector#107]
The component doesn't have keyboard support. I think that it would be interesting to implement it. The accessibility guidelines can be found at https://www.w3.org/TR/wai-aria-practices/#TreeView.

Here is an example in the Chrome Dev Tools:

![Apr-05-2020 17-47-55](https://user-images.githubusercontent.com/3165635/78503239-b167e600-7765-11ea-9e48-bdcb8da36929.gif)

--- 

Material-UI used to rely on `react-inspector` in the [Default Theme inspector](https://material-ui.com/customization/default-theme/?expand-path=$.palette.success#explore) but it was [replaced](https://github.com/mui-org/material-ui/pull/17662) so we could *dogfood* our implementation of the [TreeView component](https://material-ui.com/components/tree-view/). I wish we could go back to use react-inspector while still dogfooding our TreeView 😍(happy to work on it if needed).

![Apr-05-2020 17-56-55](https://user-images.githubusercontent.com/3165635/78503444-df015f00-7766-11ea-8af5-5268e6794afb.gif)

**Comment by rvantonisse** [comment]
I second this. Looks like last missing piece for a basic accessible tree. 

This is a link directly to the current WAI APG tree pattern: https://www.w3.org/WAI/ARIA/apg/patterns/treeview/
